theseus/core/robust_cost_function.py

```diff
@@ -80,20 +82,33 @@ def error(self) -> torch.Tensor:
 
     def weighted_error(self) -> torch.Tensor:
         weighted_error = self.cost_function.weighted_error()
-        squared_norm = torch.sum(weighted_error**2, dim=1, keepdim=True)
-        error_loss = self.loss.evaluate(squared_norm, self.log_loss_radius.tensor)
-
+        
+        if self.RobustSum:
```

Comment:
I think this can be done much more simply. If I'm not mistaken, you can keep all of the old code and just add the following lines at the beginning of `weighted_error()`
```python
if self.flatten_dims:
    weighted_error = weighted_error.view(-1, 1)
```
and then reshape the output back to `(-1, self.dim())` before returning. 

A similar procedure can be followed for `weighted_jacobians`.

Reply:
I am not sure what you mean. And why I need to reshape it. Because what I want is given any shape of error, like (Batch, Dim) and the corresponding radius with the shape of (Batch, 1), they will all be passed to the evaluation function self.loss.evaluate and return the scaled loss. 

![image](https://user-images.githubusercontent.com/52436595/235060003-008de329-5f42-4d7f-b76a-71a44632bf3f.png)

Reply:
It's a bit hard to explain, but it works, I just tested it. I'll put another PR soon.

Reply:
Please take a look at #503. The code for the jacobians part was a bit tricker than I suggested above, but I was still able to mostly use all of the old computation. Also look at the unit test I added and let me know if covers the expected behavior. Thanks!

Reply:
> It's a bit hard to explain, but it works, I just tested it. I'll put another PR soon.

Oh! I see what you mean! You treat every element of the original cost as the original "square norm"